In [ ]:
Question 1:
Create a function to find Factorial of a number using Recursion


In [2]:
def factorial(n):
   if n == 1:
       return n
   else:
       return n*factorial(n-1)

num = int(input("enter a number"))


if num < 0:
   print("Sorry, factorial does not exist for negative numbers")
elif num == 0:
   print("The factorial of 0 is 1")
else:
   print("The factorial of", num, "is", recur_factorial(num))

enter a number8
The factorial of 8 is 40320


In [ ]:
Question 2:
Create a function and display ‘n’ Fibonacci numbers using Recursion


In [5]:


def fibo(n):
   if n <= 1:
       return n
   else:
       return(fibo(n-1) + fibo(n-2))

nterms = int(input("enter a number"))


if nterms <= 0:
   print("Plese enter a positive integer")
else:
   print("Fibonacci sequence:")
   for i in range(nterms):
       print(fibo(i))

enter a number19
Fibonacci sequence:
0
1
1
2
3
5
8
13
21
34
55
89
144
233
377
610
987
1597
2584


In [ ]:
Question 3:
Print sum of list of elements without using sum() function


In [11]:
total = 0
  

list1 = [11, 5, 17, 18, 23]  
  
#
for ele in range(0, len(list1)): 
    total = total + list1[ele] 
  
print("Sum of all elements in given list: ", total) 

Sum of all elements in given list:  74


In [ ]:
Question 4:
print only even numbers from the list


In [13]:
def even_num(l):
    enum = []
    for n in l:
        if n % 2 == 0:
            enum.append(n)
    return enum
print(even_num([1, 2, 3, 4, 5, 6, 7, 8, 9]))

[2, 4, 6, 8]


In [ ]:
Question 5:
Write a Python program to get the smallest number from a list and display index of smallest element of the list


In [ ]:


list1 = [10, 20, 1, 45, 99] 

print("Smallest element is:", min(list1)) 

In [ ]:
Question 6:
Write a Python function that checks whether a passed string is palindrome or not.


In [ ]:
def isPalindrome(s): 
    return s == s[::-1]   
s = "madam"
ans = isPalindrome(s) 
  
if ans: 
    print("Yes") 
else: 
    print("No") 

In [ ]:
Question 7:
Write a Python program to count the number of even and odd numbers from a series of numbers. 
Sample numbers : numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9]
Expected Output :
Number of even numbers : 4
Number of odd numbers : 5


In [ ]:
numbers = (1, 2, 3, 4, 5, 6, 7, 8, 9) 
count_odd = 0
count_even = 0
for x in numbers:
        if not x % 2:
            count_even+=1
        else:
            count_odd+=1
print("Number of even numbers :",count_even)
print("Number of odd numbers :",count_odd)

In [ ]:
Question 8:
Write a Python program that prints all the numbers from 0 to 6 except 3 and 6.
Note : Use 'continue' statement.
Expected Output : 0 1 2 4 5


In [ ]:
for x in range(6):
    if (x == 3 or x==6):
        continue
    print(x,end=' ')
print("\n")

In [ ]:
Question 9:
Write a python function to check whether the given number is prime or not


In [4]:
num = int(input("enter a number"))
if num > 1: 
        for i in range(2, num): 
            if (num % i) == 0: 
                print(num, "is not a prime number")
                break
            else: 
                    print(num, "is a prime number") 
else: 
    print(num, "is not a prime number") 
    

enter a number8
8 is not a prime number


### Question 10:
Write a python function to check whether the given number is Adam number or not
Example: 
Input : 12
Output : Adam Number

Explanation: 12*12 = 144
Reverse of 12 is 21 → 21*21 =441
Reverse of 144 == 441


In [7]:
n=int(input())
def r(n):
    return(int(str(n) [::-1]))
def adam(n):
    if r(n)**2==r(n**2):
        return('Adam number')
    else:
        return('Not an Adam number')
print(adam(n))

12
Adam number
